sklearn/neural_network/_multilayer_perceptron.py

```diff
@@ -690,7 +695,7 @@ def _fit_stochastic(
     def _update_no_improvement_count(self, early_stopping, X_val, y_val):
         if early_stopping:
             # compute validation score, use that for stopping
-            self.validation_scores_.append(self.score(X_val, y_val))
+            self.validation_scores_.append(self._score(X_val, y_val))
```

Comment:
If we implement https://github.com/scikit-learn/scikit-learn/issues/24411#issuecomment-1247791363 and allow custom scorers, then we still need to turn off validation in `predict` during validation. We can use a similar workaround from `HistGradientBoosting*`, where it sets `_in_fit=True` in `fit` to disable validation when computing the validation scores:

https://github.com/scikit-learn/scikit-learn/blob/c557080d39cce19d944dcf19666195dfd0615401/sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py#L400

I'm okay with leaving this PR as is and dealing with https://github.com/scikit-learn/scikit-learn/issues/24411 when we get there.

What do you think?

Reply:
I would wait and deal with it in the PR that implements #24411. Mostly because who knows if we will ever get around to having a PR for that and what exactly will be implemented.